bpf/bpf_lxc.c

```diff
@@ -1476,6 +1498,19 @@ int handle_to_container(struct __ctx_buff *ctx)
 	send_trace_notify(ctx, trace, identity, 0, 0,
 			  ctx->ingress_ifindex, 0, TRACE_PAYLOAD_LEN);
 
+#if defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_ROUTING)
+	/* If the packet comes from the hostns and per-endpoint routes are enabled,
+	 * jump to bpf_host to enforce egress host policies before anything else.
+	 * We will jump back to bpf_lxc once host policies are enforced.
+	 */
+	if (identity == HOST_ID) {
+		ctx_store_meta(ctx, CB_FROM_HOST, 1);
+		ctx_store_meta(ctx, CB_DST_ENDPOINT_ID, LXC_ID);
+		tail_call_static(ctx, &POLICY_CALL_MAP, HOST_EP_ID);
```

Comment:
How do we make sure that we bypass this once bpf_host does the jump-back?

Reply:
`identity` won't be `HOST_ID` anymore since we inherit that identity from the packet mark via the iptables rule and clear it in `inherit_identity_from_host`. I'll add a comment to clarify that in the follow up.